examples/ramen_rec/example_run.ipynb

```diff
@@ -2,35 +2,38 @@
  "cells": [
```

Comment:
The constraint solver should be much more encapsulated than this example shows. If you have to change the example when the constraint solver changes then the code is too low-level. The solver should be a "filter" stage in the framework, and the input should be at the domain level, not at the level of the solver's API. That means figuring out how to encode some of the structure the solver needs into the candidate models (or have candidate set / etc.) I'm too far behind on details to recommend specific changes, I'm just seeing the layering violation.

Reply:
I've tried moving the constraint solver stuff to be a "generator" stage, and made a new pipeline for this process of recommending a fake "meal plan". The new pipeline calls one of the old pipelines to generate a bunch of Ramen candidates, then feeds those candidates in to generate a meal plan.